developers/agents/query.md

```diff
@@ -14,35 +14,61 @@ import PyCode from '!!raw-loader!/developers/agents/_includes/query_agent.py';
 
+<!-- TODO: do we need this sentece? -->
+<!-- The resulting workflow for the developer looks as follows:  -->
+
+<!-- TODO: update the image -->
+<!-- remove (3)/(6) -->
+<!-- simplify output, remove Searches executed..., only keep the first 2 lines -->
 ![Weaviate Query Agent from a user perspective](./_includes/query_agent_usage.png "Weaviate Query Agent from a user perspective")
 
 ## Architecture
 
-:::tip Details for the curious
+<!-- TODO: do we need this tip? if yes, can we make it a one-liner? -->
+<!-- :::tip Details for the curious
 You do not need to know the below technical details to use the Query Agent. The Query Agent simply returns an answer to the user's query.
 <br/>
 
 The below details are provided for the curious minds who want to know more about the inner workings of the Agent.
-:::
+::: -->
 
 The Query Agent is provided as a service on Weaviate Cloud.
 
-When a user provides a query, the query agent analyses it and any other known context to autonomously carry out the searches itself. The context may include what is known about the available Weaviate collections, previous conversation history, and any other relevant information.
+<!-- TODO: thoughts? -->
+<!-- When a user provides a query, the query agent analyses it and any other known context to autonomously carry out the searches itself. The context may include what is known about the available Weaviate collections, previous conversation history, and any other relevant information. -->
+When a user provides a prompt/query, the query agent analyses it and any other known context to autonomously carry out the searches itself.
+
+<!-- TODO: what do you think of this updated tip? -->
+:::tip Query Agent context
+The Query agent analyses collection and property descriptions to better undestand how to construct relevant queries.<br/>The context may also include previous conversation history, and any other relevant information.
+:::
+
+
+<!-- TODO: thoughts on adding the below anchor? -->
+## Query Agent in steps
 
-A high-level view of the Query Agent is shown below:
+<!-- TODO: the felt a bit redundant, so I commented it out. -->
+<!-- A high-level view of the Query Agent is shown below: -->
 
 ![Weaviate Query Agent at a high level](./_includes/query_agent_architecture.png "Weaviate Query Agent at a high level")
 
-The Query Agent works as follows at a high level (the actual implementation may vary):
+<!-- The Query Agent works as follows at a high level (the actual implementation may vary): -->
+The Query Agent runs in the following high-level steps (the actual implementation may vary):
 
 - Use appropriate foundation models (e.g. large language models) to analyze the task & the required queries. Determine the exact queries to perform. (Steps 1 & 2)
+<!-- TODO: question? do we need to talk about the vectorizing here? Isn't it enough to say that a query happens? Sometimes this could be a keyword search or even a fetch with a filter. -->
```

Comment:
We do say the vectorization happens "as needed". Initially added it here for people coming from outside of Weaviate but on reflection may not be needed, as it is for WCD users.